src/index.ts

```diff
@@ -7,120 +7,134 @@ metadata = {
+        var url = "https://graph.microsoft.com/v1.0/users/" + encodeURIComponent(parameters["userPrincipalName"]) + "/mailfolders%28%27Inbox%27%29/messages/" + encodeURIComponent(parameters["id"]);
+        //console.log(url);
+        xhr.open("GET", url);
 
-    // Use Service Instance OAuth configuration
-    //xhr.withCredentials = true;
+        // Use Service Instance OAuth configuration
+        //xhr.withCredentials = true;
 
-    // Use Access Token
-    xhr.setRequestHeader("Authorization", "Bearer " + encodeURIComponent(parameters["accessToken"]));
-    xhr.send();   
+        // Use Access Token
+        xhr.setRequestHeader("Authorization", "Bearer " + encodeURIComponent(parameters["accessToken"]));
+        xhr.send();
+    });
 }
 
-function onexecuteMessageList(parameters: SingleRecord, properties: SingleRecord) {
-    var xhr = new XMLHttpRequest();
+function onexecuteMessageList(parameters: SingleRecord, properties: SingleRecord): Promise<void> {
+    return new Promise<void>((resolve, reject) => {
+        var xhr = new XMLHttpRequest();
 
-    xhr.onreadystatechange = function() {
-            if (xhr.readyState !== 4) return;
-            if (xhr.status !== 200) throw new Error("Failed with status " + xhr.status);
+        xhr.onreadystatechange = function () {
+            try {
+                if (xhr.readyState !== 4) return;
+                if (xhr.status !== 200) throw new Error("Failed with status " + xhr.status);
 
-            //console.log(xhr.responseText);
-            var obj = JSON.parse(xhr.responseText);
-            postResult(obj.map(x => {
-                return {
-                    "id": x.id,
-                    "subject": x.subject,
-                    "from": x.from.emailAddress.address,
-                    "body": x.body.content
-                }
-            }));
-    };
+                //console.log(xhr.responseText);
+                var obj = JSON.parse(xhr.responseText);
+                postResult(obj.map(x => {
+                    return {
+                        "id": x.id,
+                        "subject": x.subject,
+                        "from": x.from.emailAddress.address,
+                        "body": x.body.content
+                    }
+                }));
+                resolve();
+            } catch (error) {
+                reject(error);
+            }
+        };
 
-    // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map
-    var url = "https://graph.microsoft.com/v1.0/users/" + encodeURIComponent(parameters["userPrincipalName"]) + "/mailfolders%28%27Inbox%27%29/messages";
-    //console.log(url);
-    xhr.open("GET", url);
+        // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map
+        var url = "https://graph.microsoft.com/v1.0/users/" + encodeURIComponent(parameters["userPrincipalName"]) + "/mailfolders%28%27Inbox%27%29/messages";
+        //console.log(url);

```

Comment:
Should we remove

Reply:
Yes

Reply:
Well I was not looking at the functionality or code within the methods. What excatly needs to be removed? the commented code?

Reply:
yeah just the console.log

Reply:
done